client/consensus/aura/src/lib.rs

```diff
@@ -302,6 +302,29 @@ impl<H, B, C, E, I, P, Error, SO> slots::SimpleSlotWorker<B> for AuraWorker<C, E
 			consensus_common::Error::ClientImport(format!("{:?}", e)).into()
 		})
 	}
+
+	fn proposing_remaining_duration(
+		&self,
+		_head: &B::Header,
+		slot_info: &SlotInfo
+	) -> Option<std::time::Duration> {
+		// never give more than 20 times more lenience.
+		const BACKOFF_CAP: u64 = 20;
+
+		let slot_remaining = self.slot_remaining_duration(slot_info);
+		let parent_slot = slot_info.last_number;
+
+		// we allow a lenience of the number of slots since the head of the
+		// chain was produced, minus 1 (since there is always a difference of at least 1)
+		//
+		// linear back-off.
+		// in normal cases we only attempt to issue blocks up to the end of the slot.
+		// when the chain has been stalled for a few slots, we give more lenience.
+		let slot_lenience = slot_info.number.saturating_sub(parent_slot + 1);
```

Comment:
From the looks of it, slot_lenience will always be 0 until it wraps around?

Reply:
I think `parent_slot` here isn't right; it's the most recent slot number that the slot worker tried to author on (which will always be ~`n_authorities` slots ago in Aura). 

You should extract the slot from the parent header. #4020 has some code for that.

Reply:
Ok @rphmeier I presume this logic existed before I changed but its updated now. Testing soon once all this jazz builds.